### PR TITLE
You can emag grapple gun to use it on station.

### PR DESCRIPTION
**Pull request by Xackii**
## About The Pull Request

You can emag grapple gun to use it on station.

## Why It's Good For The Game

Cool hook. Gives him the way to use his mechanics at the station.

:cl:
add: You can emag grapple gun to use it on station.
/:cl:

### CODE REVIEW

File: code/modules/mining/equipment/grapple_gun.dm [hunk 1]
```diff
@@ -41,7 +41,7 @@
 	if(target == user || !hooked)
 		return NONE
 
-	if(!lavaland_equipment_pressure_check(get_turf(user)))
+	if(!(lavaland_equipment_pressure_check(get_turf(user)) || (obj_flags & EMAGGED)))
 		user.balloon_alert(user, "gun mechanism wont work here!")
```
Comment (DaCoolBoss) [inline code/modules/mining/equipment/grapple_gun.dm:45]: Might as well get this typo fixed while you're here. wont -> won't.

**Comment by moreryrator** [comment]
346946832-84c244ca-1c04-4114-b172-c7766e1bd6ce (1)

**Comment by SnakeKongar** [comment]
346950075-a94171f0-ee2e-4a71-b4d0-21f3da02275b

**Comment by Sakamoto4ka** [comment]
346946832-84c244ca-1c04-4114-b172-c7766e1bd6ce

**Comment by DrDiasyl** [comment]
Livrah would be proud

File: code/modules/mining/equipment/grapple_gun.dm [hunk 2]
```diff
@@ -41,8 +41,8 @@
 	if(target == user || !hooked)
 		return NONE
 
-	if(!lavaland_equipment_pressure_check(get_turf(user)))
-		user.balloon_alert(user, "gun mechanism wont work here!")
+	if(!(lavaland_equipment_pressure_check(get_turf(user)) || (obj_flags & EMAGGED)))
```
Comment (MrMelbert) [inline code/modules/mining/equipment/grapple_gun.dm:44]:
```suggestion
	if(!lavaland_equipment_pressure_check(get_turf(user) && !(obj_flags & EMAGGED))
```

File: code/modules/mining/equipment/grapple_gun.dm [hunk 3]
```diff
@@ -73,6 +73,13 @@
 	update_appearance()
 	return ITEM_INTERACT_SUCCESS
 
+/obj/item/grapple_gun/emag_act(mob/user, obj/item/card/emag/emag_card)
+	. = ..()
+	if(obj_flags & EMAGGED)
+		return FALSE
+	balloon_alert(user, "pressure settings overloaded")
+	obj_flags |= EMAGGED
```
Comment (MrMelbert) [inline code/modules/mining/equipment/grapple_gun.dm:81]: `return TRUE` i think

**Comment by alien3301** [comment]
It should work on the station at all times. The hook does nothing special and costs a ton of points so why make this emag only?

**Comment by LePiromano** [comment]
image

**Comment by optimumtact** [comment]
I will ban you if you spam the image

**Pull request merged**